test/utils/matchers.go

```diff
@@ -168,6 +168,57 @@ func (matcher *ExitMatcher) MatchMayChangeInTheFuture(actual interface{}) bool {
 	return true
 }
 
+// ExitCleanly asserts that a PodmanSession exits 0 and with no stderr
+func ExitCleanly() types.GomegaMatcher {
+	return &exitCleanlyMatcher{}
+}
+
+type exitCleanlyMatcher struct {
+	msg string
+}
+
+type podmanSession interface {
+	ExitCode() int
+	ErrorToString() string
+}
+
+func (matcher *exitCleanlyMatcher) Match(actual interface{}) (success bool, err error) {
+	session, ok := actual.(podmanSession)
+	if !ok {
+		return false, fmt.Errorf("ExitCleanly must be passed a PodmanSession; Got:\n %+v\n%q", actual, format.Object(actual, 1))
+	}
+
+	exitcode := session.ExitCode()
+	stderr := session.ErrorToString()
+	if exitcode != 0 {
+		matcher.msg = fmt.Sprintf("Command failed with exit status %d", exitcode)
+		if stderr != "" {
+			matcher.msg += ". See above for error message."
+		}
+		return false, nil
+	}
+
+	// FIXME: #19809, "failed to connect to syslog" warnings on f38
+	// FIXME: so, until that is fixed, don't check stderr if containerized
+	if !Containerized() {
```

Comment:
Given:
1. I can't figure out where that warning is coming from; and
2. enabling _some_ testing is better than what we have now

...I've decided to kludge in a manual exception to skip warn-testing when running in container. 

@containers/podman-maintainers please take another look